src/Orleans.Runtime/Utilities/UnobservedExceptionsHandler.cs

```diff
@@ -0,0 +1,90 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Orleans.Runtime.Scheduler;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Orleans.Runtime
+{
+    internal static class ServiceProviderExtensions
+    {
+        public static void InitializeSiloUnobservedExceptionsHandler(this IServiceProvider services)
+        {
+            //resolve handler from DI to initialize it
+            var ignore = services.GetService<SiloUnobservedExceptionsHandler>();
+        }
+    }
+
+    internal class SiloUnobservedExceptionsHandler : IDisposable
+    {
+        private readonly ILogger logger;
+        
+        public SiloUnobservedExceptionsHandler(ILogger<SiloUnobservedExceptionsHandler> logger)
+        {
+            this.logger = logger;
+            AppDomain.CurrentDomain.UnhandledException += this.DomainUnobservedExceptionHandler;
+            TaskScheduler.UnobservedTaskException += InternalUnobservedTaskExceptionHandler;
+        }
+
+        private void InternalUnobservedTaskExceptionHandler(object sender, UnobservedTaskExceptionEventArgs e)
+        {
+            var aggrException = e.Exception;
+            var baseException = aggrException.GetBaseException();
+            var tplTask = (Task)sender;
+            var contextObj = tplTask.AsyncState;
+            var context = contextObj as ISchedulingContext;
+
+            try
+            {
+                UnobservedExceptionHandler(context, baseException);
+            }
+            finally
+            {
+                if (e.Observed)
+                {
+                    logger.Info(ErrorCode.Runtime_Error_100311, "UnobservedExceptionsHandlerClass caught an UnobservedTaskException which was successfully observed and recovered from. BaseException = {0}. Exception = {1}",
+                            baseException.Message, LogFormatter.PrintException(aggrException));
+                }
+                else
+                {
+                    var errorStr = String.Format("UnobservedExceptionsHandlerClass Caught an UnobservedTaskException event sent by {0}. Exception = {1}",
+                            OrleansTaskExtentions.ToString((Task)sender), LogFormatter.PrintException(aggrException));
+                    logger.Error(ErrorCode.Runtime_Error_100005, errorStr);
+                    logger.Error(ErrorCode.Runtime_Error_100006, "Exception remained UnObserved!!! The subsequent behavior depends on the ThrowUnobservedTaskExceptions setting in app config and .NET version.");
```

Comment:
@sergeybykov, should we just remove this alarming error altogether? This might have been important before, but the default now is safe, and it would only crash if the developer explicitly turned that knob on because he knew what he was doing

Reply:
I don't know the context on that. But @sergey do want to make configuring UnObservedExceptionHandler optional. I guess we can achieve that by adding an extension method on IServiceCollection or IServiceProvider to op in UnobservedExceptionsHandler. And silo shouldn't turn it on by default. 

regardless, can we do that in another PR? this PR is really just about making UnObservedExceptionHandler non-static. Not making NonObservedExceptionHandler perfect.

Reply:
Sure, we can address later. BTW, I didn't mean remove the entire thing, just the last very alarming error, which is not useful at all.
BTW, let's sync up later, as I don't think it's that critical to make it optional now... it was mainly because it was noise especially in the client, but not in the silo.